test/e2e/dra/dra.go

```diff
@@ -2646,6 +2736,29 @@ func (b *builder) setUp(ctx context.Context) {
 	b.podCounter = 0
 	b.claimCounter = 0
 	b.create(ctx, b.class())
+	// DRAExtendedResource requires scheduler to have 'create' permission on resourceclaim.
+	role, err := b.f.ClientSet.RbacV1().ClusterRoles().Get(ctx, "system:kube-scheduler", metav1.GetOptions{})
+	framework.ExpectNoError(err, "get kube-scheduler cluster role")
+	var rules []rbacv1.PolicyRule
+	for _, rule := range role.Rules {
+		if len(rule.Resources) == 1 && rule.Resources[0] == "resourceclaims" {
+			rule.Verbs = append(rule.Verbs, "create")
+		}
+		rules = append(rules, rule)
+	}
+	role.Rules = rules
+	_, err = b.f.ClientSet.RbacV1().ClusterRoles().Update(ctx, role, metav1.UpdateOptions{})
+	framework.ExpectNoError(err, "update kube-scheduler cluster role to add resourceclaim create permission")
+	cns, err := b.f.ClientSet.CoreV1().Nodes().List(ctx, metav1.ListOptions{})
+	framework.ExpectNoError(err, "getting cluster nodes")
+	for _, cn := range cns.Items {
+		if _, ok := b.driver.Nodes[cn.Name]; ok {
+			continue
+		}
+		cn.Status.Capacity[v1.ResourceName(extendedResourceName)] = resource.MustParse("1")
+		_, err = b.f.ClientSet.CoreV1().Nodes().UpdateStatus(ctx, &cn, metav1.UpdateOptions{})
+		framework.ExpectNoError(err, "update node capacity")
+	}
```

Comment:
Tests run in parallel unless explicitly marked with `ginkgo.Serial`. If you need to modify the global state of the cluster, mark your tests like that (there's one example), then modify the cluster only in those tests and restore the previous state in a `ginkgo.DeferCleanup`.

Also, "UpdateStatus" may fail due to concurrent changes by the kubelet, you need a retry loop:
> update node capacity: Operation cannot be fulfilled on nodes "kind-worker2": the object has been modified; please apply your changes to the latest version and try again

Reply:
added retry loop for the update, and updateStatus.

update node.status.capacity is only needed for the mulit-node case where some nodes have the DRA driver, and other nodes have the device plugin.

setUp() and tearDown() are updated to only update node.status.capacity for multi-node test case.

also multi-node test case needs to run Serial

the single-node test case can still run in parallel with other tests, as it does not need to update the node.status.capacity.